pkgs/build-support/docker/examples.nix

```diff
@@ -382,4 +382,40 @@ rec {
     contents = pkgs.bashInteractive;
   };
 
+  # buildLayeredImage with unprivileged user
+  bashLayeredWithUser =
+  let
+    nonRootShadowSetup = { user, uid, gid ? uid }: with pkgs; [
```

Comment:
Do we need to set up `/etc/shadow` to spin up the container? AFAIK it should work without until something tries to lookup the user information, which is not the case in this test case. 

It should make the test case a lot smaller to remove `nonRootShadowSetup`.

Reply:
As you said it’s not strictly necessary but I think it’s useful to have an example of how to do it and ideally with a comment explaining when one might need it.